corporate_partner_access/services/invitations.py

```
@@ -40,6 +41,9 @@ def apply_status(
         touched = set(update_fields or [])
         old_status = invitation.status
 
+        invitation.user = get_user_model().objects.get(email=invitation.invite_email)

```

Can we send invitations to emails that don’t have a user-account created?
I understand that every user has a user.email attribute, but what if they don’t?

In those cases, I think we should use a try block to catch the DoesNotExist exception. The exception can simply be passed, allowing the workflow to continue without crashing.

---

Good poin let me clarify this:

This method is only used when forcing acceptance via the Django admin panel, CLI, or an external tool. In this context, it’s assumed that the admin already knows the user exists.

Every user in the platform is guaranteed to have an email (the User.email field is required). So there is no scenario where a user account exists without an email.

The only case where a DoesNotExist could happen is if the invitation_email doesn’t match any user. In practice, this shouldn’t occur here, but if it does, it would simply mean the admin tried to force-accept for a non-existent user. We can decide whether to explicitly catch that exception (and pass/ignore) or let it fail so the admin notices the mismatch.

What do you think?

---

Thank you so much for the clarification.

Now my doubt is resolved, I think let if fail will be enough to let admins know what happened 👍